src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+ 
+/** Stores blob keys for plant images in Datastore */
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // TODO: usernames
+    // String username = request.getParameter("username");
+ 
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    ImmutableMap<String, List<BlobKey>> blobs = ImmutableMap.copyOf(blobstoreService.getUploads(request));
+    ImmutableList<BlobKey> blobKeys = ImmutableList.copyOf(blobs.get("image"));
+    String blobKey = blobKeys.get(0).getKeyString();
+ 
+    // TODO: might need a loop in here to check if that username already has a
+    // photo and replace it
+ 
+    Entity plantImageEntity = new Entity("plantImage");
+    plantImageEntity.setProperty("timestamp", System.currentTimeMillis());
```

Comment:
How come you are adding a time property to the plant images? Is it to help when cre

Reply:
at the moment, I was using it to get the most recently uploaded image (to make sure that the pictures were uploading each time). I don't think we'll need this when we only have one picture per person, we'll just need a username attached instead of a timestamp. Now that I know it's working, should I take it out?

Reply:
Yeah, if you don't need it I would definitely say take it out

Reply:
got it! It's removed